`azurerm_kubernete_cluster` crashes after 3.29 upgrade

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

<!--- Please keep this note for the community --->

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->


### Terraform Version

1.3.2

### AzureRM Provider Version

3.29.0

### Affected Resource(s)/Data Source(s)

azurerm_kubernetes_cluster

### Terraform Configuration Files

```hcl
resource "azurerm_kubernetes_cluster" "this" {
  name                = azurecaf_name.caf_name_aks.result
  resource_group_name = var.resource_group_name
  location            = var.location

  kubernetes_version                  = var.kubernetes_version
  node_resource_group                 = "${var.resource_group_name}-nodes"
  sku_tier                            = var.sku_tier
  private_cluster_enabled             = true
  dns_prefix                          = var.disable_private_dns_zone ? azurecaf_name.caf_name_aks.result : null
  dns_prefix_private_cluster          = var.disable_private_dns_zone ? null : azurecaf_name.caf_name_aks.result
  private_dns_zone_id                 = var.disable_private_dns_zone ? "None" : local.private_dns_zone_id
  private_cluster_public_fqdn_enabled = var.disable_private_dns_zone
  local_account_disabled              = false
  role_based_access_control_enabled   = true
  azure_policy_enabled                = true
  open_service_mesh_enabled           = var.open_service_mesh_enabled
  workload_identity_enabled           = var.workload_identity_enabled
  oidc_issuer_enabled                 = true

  ...
}
```


### Debug Output/Panic Output

```shell
module.k8s.azurerm_kubernetes_cluster.this: Refreshing state... [id=/subscriptions/xxxxx/resourceGroups/rgrp-xxxxx/providers/Microsoft.ContainerService/managedClusters/xxxx]
╷
│ Error: Plugin did not respond
│ 
│   with module.k8s.azurerm_kubernetes_cluster.this,
│   on .terraform/modules/k8s/main.tf line 103, in resource "azurerm_kubernetes_cluster" "this":
│  103: resource "azurerm_kubernetes_cluster" "this" {
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin.(*GRPCProvider).ReadResource call. The plugin logs may contain more
│ details.
╵

Stack trace from the terraform-provider-azurerm_v3.29.0_x5 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x4a79642]

goroutine 460 [running]:
github.com/hashicorp/terraform-provider-azurerm/internal/services/containers.resourceKubernetesClusterRead(0xc0010aed00, {0x5a46000?, 0xc001623400})
	github.com/hashicorp/terraform-provider-azurerm/internal/services/containers/kubernetes_cluster_resource.go:1885 +0x882
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0x6c06b60?, {0x6c06b60?, 0xc0012707e0?}, 0xd?, {0x5a46000?, 0xc001623400?})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.18.0/helper/schema/resource.go:712 +0x178
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc000cb1a40, {0x6c06b60, 0xc0012707e0}, 0xc0014535f0, {0x5a46000, 0xc001623400})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.18.0/helper/schema/resource.go:1015 +0x585
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0xc000527c98, {0x6c06b60?, 0xc001270720?}, 0xc00065ff40)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.18.0/helper/schema/grpc_provider.go:613 +0x497
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadResource(0xc0002edf40, {0x6c06b60?, 0xc001270210?}, 0xc001fef440)
	github.com/hashicorp/terraform-plugin-go@v0.10.0/tfprotov5/tf5server/server.go:747 +0x41e
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler({0x6079be0?, 0xc0002edf40}, {0x6c06b60, 0xc001270210}, 0xc001fef3e0, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.10.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:349 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0002a8a80, {0x6c18d30, 0xc0011fc680}, 0xc0010b7b00, 0xc001436b10, 0xad9cd50, 0x0)
	google.golang.org/grpc@v1.47.0/server.go:1283 +0xcfd
google.golang.org/grpc.(*Server).handleStream(0xc0002a8a80, {0x6c18d30, 0xc0011fc680}, 0xc0010b7b00, 0x0)
	google.golang.org/grpc@v1.47.0/server.go:1620 +0xa1b
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	google.golang.org/grpc@v1.47.0/server.go:922 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.47.0/server.go:920 +0x28a

Error: The terraform-provider-azurerm_v3.29.0_x5 plugin crashed!
```


### Expected Behaviour

When updating to use provider version 3.29 with an existing cluster it should be able to refresh state of the existing cluster when planning.

### Actual Behaviour

During planning phase it fails with error message provided. Seems like the code on line `1885` changed to read a pointer instead of value:

```go
d.Set("public_network_access_enabled", *props.PublicNetworkAccess != managedclusters.PublicNetworkAccessDisabled)
```

### Steps to Reproduce

1. `terraform apply` / `terraform plan`

### Important Factoids

_No response_

### References

_No response_

--- a/internal/services/containers/kubernetes_cluster_resource.go
+++ b/internal/services/containers/kubernetes_cluster_resource.go
@@ -1907,7 +1907,12 @@ func resourceKubernetesClusterRead(d *pluginsdk.ResourceData, meta interface{}) 
 		d.Set("node_resource_group", props.NodeResourceGroup)
 		d.Set("enable_pod_security_policy", props.EnablePodSecurityPolicy)
 		d.Set("local_account_disabled", props.DisableLocalAccounts)
-		d.Set("public_network_access_enabled", *props.PublicNetworkAccess != managedclusters.PublicNetworkAccessDisabled)
+
+		publicNetworkAccess := "Enabled"
+		if props.PublicNetworkAccess != nil {
+			publicNetworkAccess = publicNetworkAccess
+		}
+		d.Set("public_network_access_enabled", publicNetworkAccess != string(managedclusters.PublicNetworkAccessDisabled))
 
 		upgradeChannel := ""
 		if profile := props.AutoUpgradeProfile; profile != nil && *profile.UpgradeChannel != managedclusters.UpgradeChannelNone {

--- a/internal/services/containers/kubernetes_cluster_resource.go
+++ b/internal/services/containers/kubernetes_cluster_resource.go
@@ -1908,11 +1908,11 @@ func resourceKubernetesClusterRead(d *pluginsdk.ResourceData, meta interface{}) 
 		d.Set("enable_pod_security_policy", props.EnablePodSecurityPolicy)
 		d.Set("local_account_disabled", props.DisableLocalAccounts)
 
-		publicNetworkAccess := "Enabled"
+		publicNetworkAccess := managedclusters.PublicNetworkAccessEnabled
 		if props.PublicNetworkAccess != nil {
-			publicNetworkAccess = publicNetworkAccess
+			publicNetworkAccess = *props.PublicNetworkAccess
 		}
-		d.Set("public_network_access_enabled", publicNetworkAccess != string(managedclusters.PublicNetworkAccessDisabled))
+		d.Set("public_network_access_enabled", publicNetworkAccess != managedclusters.PublicNetworkAccessDisabled)
 
 		upgradeChannel := ""
 		if profile := props.AutoUpgradeProfile; profile != nil && *profile.UpgradeChannel != managedclusters.UpgradeChannelNone {

